src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8333,6 +8376,281 @@ private StatementSyntax ParseLocalDeclarationStatement()
+        /// Never returns null.
+        /// </summary>
+        private VariableDeclarationSyntax ParseDeconstructionIdentifierOrIdentifiersParts(bool withEquals)
+        {
+            if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+            {
+                return ParseDeconstructionList(withEquals: false, justIdentifiers: true);
+            }
+            else
+            {
+                return ParseDeconstructionIdPart();
+            }
+        }
+
+        /// <summary>
+        /// Parses an individual identifier in a deconstruction declaration. For instance, in `var (id, id) = ...`
+        /// Never returns null.
+        /// </summary>
+        private VariableDeclarationSyntax ParseDeconstructionIdPart()
+        {
+            var identifier = ParseIdentifierToken();
+            var declarator = _syntaxFactory.VariableDeclarator(identifier, null, null);
+            return _syntaxFactory.VariableDeclaration(
+                                    type: null,
+                                    variables: SyntaxFactory.SeparatedList(declarator),
+                                    deconstruction: null);
+        }
+
+        /// <summary>
+        /// Parses an individual type and identifier in a deconstruction declaration. For instance, in `(type id, type id) = ...`
+        /// Never returns null.
+        /// </summary>
+        private VariableDeclarationSyntax ParseDeconstructionTypeIdPart()
+        {
+            var type = ParseType(parentIsParameter: false);
+            var identifier = ParseIdentifierToken();
+
+            var declarator = _syntaxFactory.VariableDeclarator(identifier, null, null);
+            return _syntaxFactory.VariableDeclaration(
+                                    type: type,
+                                    variables: SyntaxFactory.SeparatedList(declarator),
+                                    deconstruction: null);
+        }
+
+        /// <summary>
+        /// Check ahead for a deconstruction declaration. This requires at least one good-looking variable and the presence of an equals sign.
+        /// Doesn't move the cursor.
+        /// PROTOTYPE(tuples) Can this be done without allocations?
+        /// </summary>
+        private bool IsPossibleDeconstructionDeclaration()
+        {
+            if ((this.CurrentToken.IsVar() && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken) ||
+                this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+            {
+                var resetPoint = this.GetResetPoint();
+                try
+                {
+                    var variables = ParseDeconstructionDeclaration(withEquals: false);
+                    var equalsToken = this.EatToken(SyntaxKind.EqualsToken);
+
+                    // We need the equals token and one other confirmation that this is a deconstruction syntax
+                    return DeconstructionVariableLooksGood(variables, withEquals: false) && !equalsToken.IsMissing;
```

Comment:
The reason I didn't do that is because it spares the parsing of the expression following the equals sign. I only want to know if there is a possible deconstruction here, not fully parse it.
That said, this could be premature optimization and the code could be simplified as you suggested.